utils/jvm_dependencies.py

```diff
@@ -0,0 +1,131 @@
+import os
+import sys
+import tempfile
+import requests
+import hashlib
+import subprocess
+from dataclasses import dataclass
+
+
+@dataclass
+class Artifact:
+    group: str
+    name: str
+    version: str
+    url: str
+    bazelized_name: str
+
+
+def parse_line(line: str) -> Artifact:
+    parts = line.strip().split(':')
+    if len(parts) == 4:
+        group, name, version, _ = parts
+        group_path = group.replace('.', '/')
+        jar_name = f"{name}-{version}.jar"
+        url = f"https://repo1.maven.org/maven2/{group_path}/{name}/{version}/{jar_name}"
+        bazelized_name = name.replace('-', '_').replace('.', '_')
+        return Artifact(group, name, version, url, bazelized_name)
+
+
+def download_and_compute_sha256(artifact: Artifact) -> str:
+    # Create a temporary directory for the downloaded file
+    temp_dir = tempfile.mkdtemp()
+    file_path = os.path.join(temp_dir, artifact.name + '.jar')
+
+    try:
+        # Download the artifact
+        response = requests.get(artifact.url)
+        response.raise_for_status()
+
+        # Save the downloaded file
+        with open(file_path, 'wb') as file:
+            file.write(response.content)
+
+        # Compute the SHA-256 hash
+        sha256_hash = hashlib.sha256()
+        with open(file_path, 'rb') as file:
+            while True:
+                data = file.read(65536)  # Read in 64k chunks
+                if not data:
+                    break
+                sha256_hash.update(data)
+
+        print(f"Downloaded {artifact.name}", file=sys.stderr)
+
+        sha256 = sha256_hash.hexdigest()
+        os.remove(file_path)
+
+        return sha256
+
+    except requests.exceptions.RequestException as e:
+        print(f"Error downloading artifact: {e}")
+        return None
+
+    finally:
+        # Clean up temporary files
+        if os.path.exists(file_path):
+            os.remove(file_path)
+        if os.path.exists(temp_dir):
+            os.rmdir(temp_dir)
+
+def resolve_and_parse(artifact_str: str) -> list[Artifact]:
+    try:
+        # Run the external process using coursier resolve
+        command = ["coursier", "resolve", artifact_str]
```

Comment:
where does this binary come from? It's okay that it's not hermetic, but please include a comment so the next person can manage to get this script to run, about what they need to install first